OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```diff
@@ -73,5 +73,17 @@ public void PatrolUntil(CPos[] waypoints, LuaFunction func, int wait = 0)
 				using (var f = func.CopyReference() as LuaFunction)
 					Self.QueueActivity(new CallFunc(() => PatrolUntil(waypoints, f, wait)));
 		}
+
+		[ScriptActorPropertyActivity]
+		[Desc("Move to a cell and demolish whatever is there.")]
+		public void GoDemolish(Actor target)
+		{
+			if (Self.HasTrait<C4Demolition>() && !target.IsDead)
```

Comment:
Given I've moved it to require C4Demolition, do you mean just something like this:

if (target.IsDead) { return; }
...